File: modules/personal-chat-view/android/src/main/java/expo/modules/personalchatview/PersonalChatViewModule.kt
package expo.modules.personalchatview

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class PersonalChatViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("PersonalChatView")
    
    View(PersonalChatView::class) {
      // Text prop
      Prop("text") { view: PersonalChatView, text: String ->
        view.setText(text)
      }
      
      // Background color prop
      Prop("backgroundColor") { view: PersonalChatView, color: Int ->
        view.setContainerBackgroundColor(color)
      }
      
      // Event handlers
      Events("onPersonalChatPress")

      // Cleanup method
      AsyncFunction("cleanup") { viewTag: Int ->
        val view = appContext.findView<PersonalChatView>(viewTag)
        view?.cleanup()
      }
    }
  }
}

File: modules/personal-chat-view/ios/PersonalChatViewModule.swift
import ExpoModulesCore
import UIKit

public class PersonalChatViewModule: Module {
  public func definition() -> ModuleDefinition {
    Name("PersonalChatView")
    
    View(PersonalChatView.self) {
        // Pass object as dictionary
        Prop("chatRoomCredentials") { (view: PersonalChatView, userInfo: [String: Any]) in
            view.setChatRoomCredentials(userInfo)
        }

        Events("onSendMessage")
    }

    AsyncFunction("addReceivedMessage") { [weak self] (viewRef: Int, message: [String: Any]) -> Void in
        DispatchQueue.main.async {
            guard let view = self?.appContext?.findView(withTag: viewRef, ofType: PersonalChatView.self) else {
                return
            }
            view.addReceivedMessage(message)
        }
    }

    AsyncFunction("configure") { (supabaseURL: String, supabaseKey: String, accessToken: String, refreshToken: String) in
      try await SupabaseManager.shared.configure(supabaseURL: supabaseURL, supabaseKey: supabaseKey, accessToken: accessToken, refreshToken: refreshToken)
    }
  }
}
